### DISCUSSION TITLE
xDrip+ color settings

### DISCUSSION THREAD

**Discussion by lz2jvc** [NightscoutFoundation/xDrip#3608] [General]
Please adjust the color setting that corresponds to the bolus insulin dose reading so that it is a visible color by default. Currently, by default, it is made with a dark green color that blends into the background of the xDrip+ graphics. After I already adjusted the color to the maximum light green tone, the bolus insulin graph became visible. The first photo is before the change as it is set by default, and the second photo is after my intervention and change.
![1688659182437](https://github.com/user-attachments/assets/40ad53c0-5505-4279-82bd-dc30d443bdce)
![2024-08-04 01 15 26](https://github.com/user-attachments/assets/36ef2eaa-e061-41f7-9cf7-35813070f663)

**Comment by Navid200** [comment]
The ability to change that color exists and you are using it, right?
You just suggest that we change the default color so that you don't have to adjust it every time you install xDrip, right?  
 
A color can be a variety of colors.  How can we argue which one of thousands of possible colors is the right one for the majority?  
The problem is that a color that is best for you may be worst for another user.  
It may be different if we were talking about a binary setting.  If a switch makes more sense for the majority to be on versus off, it may make sense to argue that the default should be on.  But, a color is different.  

One can even change the background color from black to something else.  Then, everything will look different.  

What makes all of this even worse is that this is not something new.  Users have already been using xDrip for a long time with this color as default.  If we change the default color now, we are taking the chance of many people being confused and complain.

You can save your settings and every time you install xDrip, it will automatically be set after you restore your saved settings.

**Reply by lz2jvc** [reply]
I disagree with your given answer. I agree that we can always change the colors to our liking, but that's not the idea here. In order for everything in xDrip+ to be transparent and visible, both the default background must be dark, and the default color tone must be selected for the corresponding dark background to be normally visible. I don't want you to change the colors themselves, but to default to a light green color for bolus insulin. The lightest green color is not available as another option, and it stands out very well against the dark graphics of the xDrip+.  Naturally, this background color should not overlap with the other green tones used for the other lines and readings. By default, all lines should be clear, legible and visible in the default dark mode set by you on the background of xDrip+. Then anyone who decides can change the color range as he wishes according to his personal perversion. I think I was clearer in my statement now?

**Reply by Navid200** [reply]
It wasn't an answer.  What I posted was my view.
I understood your meaning the first time.  

What you have described is what you prefer.  Someone else may wish for this particular item not to stand out.  
I can assume that the developer who added this item had the latter preference.  

We can change the default now, which will impose a change for every user even those who are happy with it as is.  

Changing a default in xDrip after the fact is not easy.  I mean even if I go ahead and open a submission, it will very likely not be approved.

As an example, we have had an open issue for 6 years about the forecast low alert text being red over black, which has a very low contrast:
https://github.com/NightscoutFoundation/xDrip/issues/682  
I want to add the ability to edit that color.  Currently, the user cannot edit that color.  But, it is impossible for us to the change the default color.  After we succeed and add the feature so that the color for that text can be changed by the user if desired, I'm almost certain we will not change the default and it will remain red on black.

**Reply by lz2jvc** [reply]
Run it as a query and option to change only the color tone for the bolus insulin. Everything else is OK in my opinion, as well as the one about the red color for moving low blood sugar. But here at this stage nothing is visible except the dark background of xDrip+. This is not my whim. I assume that many people who increase insulin doses do not know about this indication that can be seen on the chart. That is, it comes out but is not visible. I also didn't know it existed until the end, until a friend of mine told me how to make it visible. I always thought that the only information about the doses of bolus insulin xDrip+ is given only above together with basalt insulin. The feature is very useful but also very hidden with this color choice.
